core/domain/collection_services_test.py

```diff
@@ -913,50 +916,46 @@ def test_update_collection_tags(self):
                     'new_value': ['duplicate', 'duplicate']
                 }], 'Add a new tag')
 
-    def test_update_collection_node_prerequisite_skills(self):
-        # Verify initial prerequisite skills are empty.
+    def test_add_and_delete_collection_skills(self):
+        # Verify skills table is initially empty.
         collection = collection_services.get_collection_by_id(
             self.COLLECTION_ID)
-        collection_node = collection.get_node(self.EXPLORATION_ID)
-        self.assertEqual(collection_node.prerequisite_skills, [])
+        self.assertEqual(collection.skills, {})
 
-        # Update the prerequisite skills.
+        # Add a skill.
         collection_services.update_collection(
             self.owner_id, self.COLLECTION_ID, [{
-                'cmd': collection_domain.CMD_EDIT_COLLECTION_NODE_PROPERTY,
-                'exploration_id': self.EXPLORATION_ID,
-                'property_name': 'prerequisite_skills',
-                'new_value': ['first', 'second']
-            }], 'Changed the prerequisite skills of a collection node')
+                'cmd': collection_domain.CMD_ADD_COLLECTION_SKILL,
+                'name': 'test'
+            }], 'Add a new skill')
 
-        # Verify the prerequisites are different.
+        # Verify that the skill is added.
         collection = collection_services.get_collection_by_id(
             self.COLLECTION_ID)
-        collection_node = collection.get_node(self.EXPLORATION_ID)
-        self.assertEqual(
-            collection_node.prerequisite_skills, ['first', 'second'])
+        self.assertEqual(collection.skills.keys(), ['skill0'])
 
-    def test_update_collection_node_acquired_skills(self):
-        # Verify initial acquired skills are empty.
-        collection = collection_services.get_collection_by_id(
-            self.COLLECTION_ID)
-        collection_node = collection.get_node(self.EXPLORATION_ID)
-        self.assertEqual(collection_node.acquired_skills, [])
+        # Verify that error will be thrown when duplicate skill is added.
```

Comment:
Having specific tests here is preferable versus one long test testing a variety of conditions.